### PR TITLE
Pilip/identity

### PR DESCRIPTION
**Pull request by blastrock**

### CODE REVIEW

File: identity/generate.go [hunk 1]
```diff
@@ -75,22 +132,20 @@ func generateToken(trustchainID []byte, trustchainPrivateKey []byte,
...
+		Target:                       "user",
+		Value:                        userID,
 		UserSecret:                   userSecret,
+		TrustchainID:                 trustchainID,
 	}
 
-	// Note: []byte values are encoded as base64-encoded strings
-	//       (see: https://golang.org/pkg/encoding/json/#Marshal)
-	jsonToken, err4 := json.Marshal(delegationToken)
+	b64Token, err4 := toB64JSON(identity)
```
Comment (AlexandreBossard) [inline identity/generate.go]: s/err4/err/

File: identity/generate.go [hunk 2]
```diff
@@ -38,6 +38,47 @@ func genericHash(input []byte) []byte {
...
+}
+
+type PublicIdentity struct {
+	TrustchainID []byte `json:"trustchain_id"`
+	Target       string `json:"target"`
+	Value        []byte `json:"value"`
+}
+
 //Generate a user token for given user.
 func Generate(config Config, userID string) (string, error) {
```
Comment (AlexandreBossard) [inline identity/generate.go]: It is called "createIdentity" in all other implementations.

File: identity_test/generate_test.go [hunk 3]
```diff
@@ -5,64 +5,123 @@ import (
...
+	It("parse a valid public identity", func() {
+		sidentity, err := base64.StdEncoding.DecodeString(goodPublicIdentity)
+		Expect(err).To(Not(HaveOccurred()))
+		publicIdentity := identity.PublicIdentity{}
+		err = json.Unmarshal(sidentity, &publicIdentity)
+		Expect(err).To(Not(HaveOccurred()))
+
+		Expect(publicIdentity.TrustchainID).To(Equal(trustchainID))
+		Expect(publicIdentity.Target).To(Equal("user"))
+		Expect(publicIdentity.Value).To(Equal(obfuscatedUserID))
```
Comment (AlexandreBossard) [inline identity_test/generate_test.go:125]: You should add error cases, and see what is returned back to the user.

**Comment by blastrock** [comment]
Moved to TankerHQ/identity-go#1

**Pull request closed without merging**